test/e2e/ownerrefs_finalizers_test.go

```diff
@@ -218,13 +265,35 @@ var (
 )
 
 // vSphereFinalizers maps VSphere infrastructure resource types to their expected finalizers.
-var vSphereFinalizers = map[string]func(types.NamespacedName) []string{
-	"VSphereVM":              func(_ types.NamespacedName) []string { return []string{infrav1.VMFinalizer} },
-	"VSphereClusterIdentity": func(_ types.NamespacedName) []string { return []string{infrav1.VSphereClusterIdentityFinalizer} },
-	"VSphereDeploymentZone":  func(_ types.NamespacedName) []string { return []string{infrav1.DeploymentZoneFinalizer} },
-	"VSphereMachine":         func(_ types.NamespacedName) []string { return []string{infrav1.MachineFinalizer} },
-	"IPAddressClaim":         func(_ types.NamespacedName) []string { return []string{infrav1.IPAddressClaimFinalizer} },
-	"VSphereCluster":         func(_ types.NamespacedName) []string { return []string{infrav1.ClusterFinalizer} },
+var vSphereFinalizers = func() map[string]func(types.NamespacedName) []string {
+	if testMode == SupervisorTestMode {
+		return map[string]func(types.NamespacedName) []string{
+			"VirtualMachine": func(_ types.NamespacedName) []string {
```

Comment:
Sorry I think I still don't understand why we are defining finalizers for VirtualMachine. We are not using this information after FilterObjectsWithKindAndName is filtering out VirtualMachines

Happy to take a look via Zoom together

Reply:
I think I got your point now, PTAL